js/src/modals/AddAddress/addAddress.js

```diff
@@ -28,6 +28,7 @@ export default class AddAddress extends Component {
 
   static propTypes = {
     contacts: PropTypes.object.isRequired,
+    address: PropTypes.string,
```

Comment:
I agree that it should be split into 3 categories. In the meanwhile, I prefer splitting in at least 2.